### PR TITLE
WT-9189 Compile CMake with RelWithDebInfo build type

### PR DESCRIPTION
**Pull request by etienneptl**
We can now build CMake using the build type RelWithDebInfo:

```
-DCMAKE_BUILD_TYPE=RelWithDebInfo
```
Additional changes were made to remove the default optimization level set by the RelWithDebInfo build type to use our own `CC_OPTIMIZE_LEVEL` instead. By default, `CMAKE_C_FLAGS_RELWITHDEBINFO` is evaluated to `-O2 -g -DNDEBUG`.

Note: This is incompatible with the `ENABLE_STRICT` flag, see WT-9209.

### CODE REVIEW

**Review by korteland — APPROVED** [review]
LGTM!

File: cmake/configs/base.cmake [hunk 1]
```diff
@@ -248,7 +248,7 @@ config_string(
     DEFAULT "\"${WT_VERSION_STRING}\""
 )
 
-if(HAVE_DIAGNOSTIC AND (NOT "${CMAKE_BUILD_TYPE}" STREQUAL "Debug"))
+if(HAVE_DIAGNOSTIC AND (NOT "${CMAKE_BUILD_TYPE}" STREQUAL "Debug" AND NOT "${CMAKE_BUILD_TYPE}" STREQUAL "RelWithDebInfo"))
```
[the remark this answers is not in the document]
Comment (etienneptl) [reply cmake/configs/base.cmake]: Very good idea, added in 864255803b242e6625ccb78aa338104b736eb911, thanks!

**Review by ajmorton — APPROVED** [review]
LGTM 👍

**Review by fedorova — APPROVED** [review]
I tested this and it produces the right flags: -O3 -g, and the resulting binary has the debug symbols.

**Pull request merged**